src/DynamoCore/Utilities/LuceneSearchUtility.cs

```diff
@@ -82,28 +120,40 @@ internal void InitializeLuceneConfig(string dirName, LuceneStorage storageType =
 
 
             // Create an analyzer to process the text
-            Analyzer = CreateAnalyzerByLanguage(dynamoModel.PreferenceSettings.Locale);
+            Analyzer = CreateAnalyzerByLanguage(PreferenceSettings.Instance.Locale);
+
+            // Check if Lucene index file exists, if not create it
+            if (!DirectoryReader.IndexExists(indexDir))
+            {
+                CreateLuceneIndex();    
+            }
+        }
 
+        internal void CreateLuceneIndex()
+        {
             // Initialize Lucene index writer, unless in test mode or we are using RAMDirectory for indexing info. 
-            if (!DynamoModel.IsTestMode || currentStorageType == LuceneStorage.RAM)
+            if (!DynamoModel.IsTestMode || startConfig.StorageType == LuceneStorage.RAM)
```

Comment:
Do we use the same indexDir path when it is in RAM mode also?

Reply:
See line 112, different `indexDir` for each storage mode